app/Main.hs

```diff
@@ -10,16 +10,81 @@
 module Main ( main ) where 
 
 import System.IO
+import Data.Char (toUpper)
 
--------------------------------------------------------------------------------
+-- Temporary Definitions - When Card/Deck are defined in the library this will be
+-- removed
+type Card = String
+type Deck = [Card]
+
+loadDeck :: IO (Maybe Deck)
+loadDeck = undefined
+
+saveDeck :: Deck -> IO ()
+saveDeck = undefined
+
+createNewDeck :: IO Deck
+createNewDeck = undefined
+
+editCurrentDeck :: Deck -> IO Deck
+editCurrentDeck = undefined
+
+play :: Deck -> IO ()
+play = undefined
+
+-- | This function is used to ensure options that require a deck are not called
+--   without a deck. This is better than checking a Maybe Deck is not Nothing at 
+--   the start of all of these options.
+requireDeck :: Maybe Deck -> (Deck -> IO ()) -> IO ()
+requireDeck deck f =
+    case deck of 
+        Nothing -> do
+            putStrLn "This option requires a deck to be selected"
+            menu Nothing
+        Just d -> f d
 
+-- | This optionally takes a Deck and shows the user a menu of available options
+--   then runs the relavant function. Keeps running until the user uses the quit 
+--   option
+menu :: Maybe Deck -> IO ()
+menu deck = do
+    -- Determines whether to show options that require a deck to be present
+    let deckOptions = if deck == Nothing then 
+                          ""
+                      else
+                          "\tS) Save Deck\n\tE) Edit Current Deck\n\tP) Play"
```

Comment:
Something like this may be nicer (see in conjunction with the next comment)?
```suggestion
    let deckOptions | deck == Nothing = []
                    | otherwise = ["\tS) Save Deck", "\tE) Edit Current Deck", "\tP) Play"]
```